### ISSUE TITLE
Upload event data to a IRS lite plan

### ISSUE THREAD

**Issue by Naima-Bashir** [onaio/reveal-frontend#1504]
https://smartregister.atlassian.net/browse/RVL-1461

The client would like to have a template which they can populate with values in and then use that to post events data to an IRS lite plan in CSV format.

Does this fall under new features or? @moshthepitt 

Some more updates: 
The client wants to be able load a fully completed implementation dataset into a newly created plan and for this data to reflect within the dashboard, this data can be data from a previous implementation plan downloaded as CSV and modified and then re uploaded to a new plan via the provided template, and this data should be viewable on dashboards when the new plan with the uploaded data is opened.

provide a template for uploading event data to a new plan.

a way of uploading that CSV data to a new plan.

step 1: Create a new lite plan

step 2: populate provided template

step 3: upload mass data once-off

step 4: visualize on dashboard

**Comment by moshthepitt** [comment]
I can't tell what is being asked for.  Can we get more details please?  let's try and answer these questions:  if this is done and delivered to the client, what would it look like?  How would they use it? From where would it be used?

**Comment by Naima-Bashir** [comment]
@moshthepitt 

The client wants to be able load a fully completed implementation dataset into a newly created plan and for this data to reflect within the dashboard, this data can be data from a previous implementation plan downloaded as CSV and modified and then re uploaded to a new plan via the provided template, and this data should be viewable on dashboards when the new plan with the uploaded data is opened.

provide a template for uploading event data to a new plan.

a way of uploading that CSV data to a new plan.

**Comment by moshthepitt** [comment]
Probably a good idea to involve @mberg and @ukanga as this seems like an entirely new feature that we have not had on the web app so far.  That is, we have not used the web app for data collection to this extent so far.

Basically it seems like the client wants to upload a CSV document, and have the web app then create OpenSRP events.  Certainly interesting, but I'm not sure how this will work.

Looks like it might be a significant bit of work given the Reveal data model. i.e. before we can have events we most likely need tasks.  We would need to decide whether we need a new endpoint for this, or if everything will be done in the web app.

**Comment by ukanga** [comment]
This seems like a new request that will also require detailed planning and resourcing. Let's push to bizdev for now.

**Comment by ukanga** [comment]
Adding any data, the best option is submitting data via OpenSRP data entry endpoints, from tasks to events. Any aspect of automation via using OpenSRP API should suffice for now. Perhaps, can we share the OpenSRP API documentation instead, which should cover this request in the meantime?

**Comment by Naima-Bashir** [comment]
@moshthepitt This has been picked up by the client engineers. FYI

**Comment by Naima-Bashir** [comment]
I have received feedback that the Akros engineers(Stephanus) has picked it up. Just fyi

**Comment by moshthepitt** [comment]
@Naima-Bashir cool, thanks.  Will they be submitting a PR to this repo?

**Comment by moshthepitt** [comment]
Closing this
